src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -98,15 +98,21 @@ internal static void Enrich(Activity activity, NpgsqlConnector connector)
 
     internal static void ReceivedFirstResponse(Activity activity)
     {
-        if (!activity.IsAllDataRequested)
+        if (!activity.IsAllDataRequested || (NpgsqlTracingOptions.Current?.DisableFirstResponseEvent ?? false))
             return;
-
+    
         var activityEvent = new ActivityEvent("received-first-response");
         activity.AddEvent(activityEvent);
     }
 
     internal static void CommandStop(Activity activity)
     {
+        if (!(NpgsqlTracingOptions.Current?.DisableLastReadEvent ?? false))
+        {
+            var activityEvent = new ActivityEvent("received-last-response");
```

Comment:
I do not think there are any reasons to have this event, given that we already close the activity/span just below, so it's always going to be at the end of it.

Reply:
removing then